### PR TITLE
Manage refresh token

### PR DESCRIPTION
**Pull request by DJA89**
# Manage refresh token

## Description:

The access token expires quickly, we use a refresh token to get a new access token

## Notes:

* Link to Blackmarket authentication flow documentation: https://www.notion.so/rootstrap/RS-Blackmarket-682df24a33c3462e9eb6246c3424efe2?pvs=4#9f2c7160e5af45f0a98ec3d19b3ab87e
* Link to NextAuth documentation on refresh tokens: https://authjs.dev/guides/basics/refresh-token-rotation

## Original design:

No design changes

## Screenshots from this PR:

No design changes

### CODE REVIEW

**Review by manuelcheveste — APPROVED** [review]

**Review by agustinmrgo — APPROVED** [review]

File: pages/api/auth/[...nextauth].ts [hunk 1]
```diff
@@ -15,10 +15,36 @@ export const authOptions: AuthOptions = {
...
+            {
+              headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+              body: new URLSearchParams({
+                refresh: token.user.refreshToken,
+              }),
+              method: 'POST',
+            }
+          );
+
+          const tokens = await response.json();
```
Comment (FlaviaCohen) [inline pages/api/auth/[...nextauth].ts:32]: I think this await is not necessary, you already have the await in line 21 and tokens is going to wait until response is resolved.

File: pages/api/auth/[...nextauth].ts [hunk 2]
```diff
@@ -15,10 +15,36 @@ export const authOptions: AuthOptions = {
...
+          if (!response.ok) throw tokens;
+
+          return {
+            ...token,
+            user: { ...token.user, accessToken: tokens.access },
+            exp: new Date(tokens.access_token_expiration).getTime(),
+          };
+        } catch (error) {
+          // Return previous token on error
+          console.log(error);
```
Comment (FlaviaCohen) [inline pages/api/auth/[...nextauth].ts]: Can you return de error or some error message instead of login it? Something more useful for the user. WDYT?

**Review by FlaviaCohen — APPROVED** [review]

File: pages/api/auth/[...nextauth].ts (hunk 2 above)
Comment (DJA89) [reply pages/api/auth/[...nextauth].ts]: This is ran on the server, so the user won't actually see anything, even the `console.log`s are only shown on my terminal and not on the browser's console. 

I don't really understand NextAuth all that much, but I believe whatever I return here is just saved as the token by NextAuth.

File: pages/api/auth/[...nextauth].ts (hunk 1 above)
Comment (DJA89) [reply pages/api/auth/[...nextauth].ts:32]: Apparently `response.json()` returns a promise, it doesn't work if I don't await it.

**Pull request merged**